src/common/providers/https.ts

```diff
@@ -607,23 +628,48 @@ export async function checkAuthToken(
 }
 
 /** @internal */
-async function checkAppCheckToken(req: Request, ctx: CallableContext): Promise<TokenStatus> {
-  const appCheck = req.header("X-Firebase-AppCheck");
-  if (!appCheck) {
+async function checkAppCheckToken(
+  req: Request,
+  ctx: CallableContext,
+  options: CallableOptions
+): Promise<TokenStatus> {
+  const appCheckToken = req.header("X-Firebase-AppCheck");
+  if (!appCheckToken) {
     return "MISSING";
   }
   try {
-    let appCheckData;
+    let appCheckData: AppCheckData;
     if (isDebugFeatureEnabled("skipTokenVerification")) {
-      const decodedToken = unsafeDecodeAppCheckToken(appCheck);
+      const decodedToken = unsafeDecodeAppCheckToken(appCheckToken);
       appCheckData = { appId: decodedToken.app_id, token: decodedToken };
+      if (options.consumeAppCheckToken) {
+        appCheckData.alreadyConsumed = false;
+      }
     } else {
-      appCheckData = await getAppCheck(getApp()).verifyToken(appCheck);
+      const appCheck = getAppCheck(getApp());
+      if (options.consumeAppCheckToken) {
+        if (appCheck.verifyToken.length === 1) {
```

Comment:
I might be wrong here but if `appCheck.verifyToken` is `undefined`, accessing `.length` throws, doesn't it?
Would it be better to do the following instead?

```
if (typeof appCheck.verifyToken === "function") { 
```

Reply:
I just realized that you are checking the number of parameters here. If we are confident that the Admin SDK used always has the appCheck API, please ignore my comment above.

Reply:
Think defensive coding is better here!